src/permutat.h

```diff
@@ -23,12 +23,74 @@
 *F  DEG_PERM4(<perm>) . . . . . . . . . . . . . degree of (large) permutation
 *F  ADDR_PERM4(<perm>)  . . . . . . . absolute address of (large) permutation
 */
-#define NEW_PERM2(deg)          NewBag( T_PERM2, (deg) * sizeof(UInt2))
-#define DEG_PERM2(perm)         (SIZE_OBJ(perm) / sizeof(UInt2))
-#define ADDR_PERM2(perm)        ((UInt2*)ADDR_OBJ(perm))
-#define NEW_PERM4(deg)          NewBag( T_PERM4, (deg) * sizeof(UInt4))
-#define DEG_PERM4(perm)         (SIZE_OBJ(perm) / sizeof(UInt4))
-#define ADDR_PERM4(perm)        ((UInt4*)ADDR_OBJ(perm))
+
+static inline UInt SIZEBAG_PERM2(UInt deg) {
+  return sizeof(Obj) + deg*sizeof(UInt2);
+}
+
+static inline Obj NEW_PERM2(UInt deg) {
+  return NewBag( T_PERM2, SIZEBAG_PERM2(deg));
+}
+
+static inline UInt DEG_PERM2(Obj perm)  {
+  return (SIZE_OBJ(perm) - sizeof(Obj)) / sizeof(UInt2);
+}
+
+static inline UInt2 *ADDR_PERM2(Obj perm) {
```

Comment:
The formatting of most of these functions does not match our `clang-format` guidelines, as the `{` is not on its own line